Bhakti Budiman Novanda

bhaktitud opened the pull request.

bhaktitud committed:
KANBAN SERVER

- google OAuth
- error handler
- authentication
- CRUD Task

bhaktitud committed:
last commit

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
package json edited

bhaktitud committed:
heroku redeploy

bhaktitud committed:
heroku redeploy

bhaktitud committed:
Update README.md

bhaktitud committed:
Update README.md

dhimashary commented on controllers/taskController.js:
```diff
@@ -0,0 +1,117 @@
+        })
+        .then((taskFound) => {
+            let task = {}
+            let backlog = []
```
hal ini baiknya di handle di client aja ya, kan ada computed yang bisa filtering tasknya

dhimashary commented on middlewares/authentication.js:
```diff
@@ -0,0 +1,43 @@
+        })
+        .then((userFound) => {
+            if(userFound){
+                req.headers.userId = userFound.dataValues.id
```
ini jangan ditaro di headers lagi ya, langsung aja req.currentUserID

dhimashary commented on routes/index.js:
```diff
@@ -0,0 +1,31 @@
+router.use(authentication)
+router.get('/task', TaskController.fetchTask)
+router.post('/task', TaskController.createTask)
+router.delete('/task/:id', TaskController.deleteTask)
```
kok gak ada authorisasi?

bhaktitud replied:
kemarin punya asumi kalau si kanban nya hanya bisa di lihat oleh orang yang punya nya aja, jd gak ada kemungkinan bisa lihat atau edit data orang lain. makanya kepikiran gak pakai authorization (hak ubah).

bhaktitud replied:
sudah di buat author

dhimashary commented on routes/index.js:
```diff
@@ -0,0 +1,31 @@
+router.get('/task', TaskController.fetchTask)
+router.post('/task', TaskController.createTask)
+router.delete('/task/:id', TaskController.deleteTask)
+router.put('/task/:id', TaskController.editTask)
```
kok gak ada authorisasi?

bhaktitud replied:
sudah di buat author, mohon feedback nya

dhimashary commented on models/user.js:
```diff
@@ -0,0 +1,34 @@
+  class User extends sequelize.Sequelize.Model{}
+  User.init({
+    name: DataTypes.STRING,
+    email: {
```
gak handle email unik, gak handle email valid

bhaktitud committed:
REVISI KANBAN

- Authorization added
- Edit feature works by double clicking the title
- some code refactors

bhaktitud committed:
code refactored
